sidecar/reloader/main.go

```diff
@@ -160,31 +107,82 @@ func main() {
 		}
 	}()
 
-	// plugin config
-	for _, file := range pluginConfigList {
-		fullFile := filepath.Join(pluginConfigDir, file)
-		if _, err := os.Stat(fullFile); err == nil {
-			log.Printf("add file %s to watcher\n", fullFile)
-			err = watcher.Add(fullFile)
-			if err != nil {
-				log.Fatal(err)
-			}
-			fileCheck[fullFile] = getMD5(fullFile)
+	done := make(chan bool)
+	<-done
+}
+
+func generateWatchedFileList(pluginConfigDir, licenseFilePath string) []string {
+	var fileList []string
+	if licenseFilePath != "" {
+		fileList = append(fileList, licenseFilePath)
+	}
+	pluginConfigs, err := ioutil.ReadDir(pluginConfigDir)
+	if err != nil {
+		log.Printf("read dir err: %v", err)
+		return fileList
+	}
+	for _, pluginConfig := range pluginConfigs {
+		if shouldIgnoreFile(pluginConfig.Name()) {
+			log.Printf("%s should not add to watcher", pluginConfig.Name())
+			continue
 		}
+		filePath := filepath.Join(pluginConfigDir, pluginConfig.Name())
+		fileList = append(fileList, filePath)
+	}
+	return fileList
+}
+
+func newReloaderWatcher() (*reloaderWatcher, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Printf("NewWathcher err: %v\n", err)
+		return nil, err
 	}
+	return &reloaderWatcher{
+		fileCheck: map[string][]byte{},
+		watcher:   watcher,
+	}, nil
+}
 
-	// license file
-	if _, err := os.Stat(licenseFilePath); err == nil {
-		log.Printf("add file %s to watcher\n", licenseFilePath)
-		err = watcher.Add(licenseFilePath)
-		if err != nil {
-			log.Fatal(err)
+func (r *reloaderWatcher) close() {
+	r.watcher.Close()
+}
+
+func (r *reloaderWatcher) isFileChanged(filePath string) (bool, error) {
```

Comment:
I think can be rename is function, like `updateWatched` or other. Because `isFileChanged` looks like a judgement call only, but in fact, it will update watcher, this is a side effect

Reply:
Done